fix(git-changelog): layout of commit inline authors and hmr bugs

northword opened the pull request:
- [x] Minor refactoring of useChangelog:
  - [x] get the filepath inside a composable function, no need to pass the parameter inside the component
  - [x] Returned commits with author information handled, no need to handle it further in the component.
- [x] fix the layout of commit inline authors, see below

  befor:

  ![befor](https://github.com/user-attachments/assets/a521b4a8-46bb-45c7-849f-1f9a3048e1d8)


  after:

  ![after](https://github.com/user-attachments/assets/cc09cd29-4705-41ef-b777-b44bfff709da)

- [x] Fixed changelog not displaying when starting dev (it worked after a refresh, probably because the author info was not captured initially)
- [x] Documentation for `displayAuthorsInsideCommitLine`.
- [x] Hot reloading bug: When manually refreshing a page, the changelogs and contributors on many pages become no changelogs due to not updating the cache to the client...
- [x] Hide external link icons as they break the layout:

  before:

  ![image](https://github.com/user-attachments/assets/7780e65c-6962-422c-aeb6-53a9867b437f)

  after:

  ![image](https://github.com/user-attachments/assets/426ee44a-5856-4e87-9763-dd0483d65b8e)

- [x] Disable VitePress' built-in lastUpdated, since relevant content is already provided when gitchangelog is enabled, this option is meaningless.

nekomeowww commented:
Much appriciated! Let's rollout a new version once merged.

northword commented:
Cc @lulu0119 for modifying layout of inside commit line authors.

northword commented:
Could you please tell me how I can fix the following error?

this pr, pnpm link:

```ts
import { h } from 'vue'
import type { Theme } from 'vitepress'
import DefaultTheme from 'vitepress/theme'

// CSS
import './styles/vars.css'
import './styles/custom.css'
import './styles/csl-styles.css'

// NProgress
// @ts-expect-error no types
import nprogress from 'nprogress'
import './styles/nprogress.css'

// Element Plus
// Dark mode support: https://element-plus.org/zh-CN/guide/dark-mode.html
import 'element-plus/theme-chalk/dark/css-vars.css'

// Git Changelog
// https://nolebase-integrations.ayaka.io/pages/zh-CN/integrations/vitepress-plugin-git-changelog/
import { NolebaseGitChangelogPlugin } from '@nolebase/vitepress-plugin-git-changelog/client'
import '@nolebase/vitepress-plugin-git-changelog/client/style.css'

// Custom Components
import Giscus from './components/Giscus.vue'
import DocFooter from './components/DocFooter.vue'
import SvgImage from './components/SvgImage.vue'
import Market from './components/Market.vue'

export default {
  extends: DefaultTheme,
  Layout() {
    return h(DefaultTheme.Layout, null, {
      // https://vitepress.dev/guide/extending-default-theme#layout-slots
      'doc-footer-before': () => [h(DocFooter)],
      'doc-after': () => h(Giscus),
    })
  },
  enhanceApp({ app, router }) {
    // Vue App Enhance
    app.component('SvgImage', SvgImage)
    app.component('Market', Market)

    app.use(NolebaseGitChangelogPlugin)

    // Router Enhance
    // nprogress
    if (!import.meta.env.SSR) {
      nprogress.configure({ showSpinner: false })
      const _cacheBeforeRouteChange = router.onBeforeRouteChange
      const _cacheAfterRouteChange = router.onAfterRouteChanged
      router.onBeforeRouteChange = (to) => {
        nprogress.start()
        _cacheBeforeRouteChange?.(to)
      }
      router.onAfterRouteChanged = (to) => {
        _cacheAfterRouteChange?.(to)
        nprogress.done()
      }
    }
  },
} satisfies Theme

```

![image](https://github.com/user-attachments/assets/45953ab8-b368-4708-a89c-eac2647fb5eb)

nekomeowww commented:
> Could you please tell me how I can fix the following error?

- What is the resolved Vue version inside of this module?
- What is the resolved Vue version inside of the VitePress module?
- What is the resolved Vue version inside of the NolebaseGitChangelog module?
- Are you using `pnpm link`?
- What was the child dependencies look like for `@nolebase/vitepress-plugin-git-changelog`? What version of `vue` is it?

I remember that I have installed the `vue` as `dependencies` as workspace dependencies, which may result in such issue where pnpm must handle and calculate the correct linked `dependencies` of `vue` for you, and I frankly not sure about how `pnpm link` was done for workspace / monorepo module.

northword commented:
> What is the resolved Vue version inside of this module?    
                   
3.4.31

> What is the resolved Vue version inside of the VitePress module?          
    
3.4.31

> What is the resolved Vue version inside of the NolebaseGitChangelog module?   

3.4.30

> Are you using pnpm link?                               
                       
yes 

> What was the child dependencies look like for @nolebase/vitepress-plugin-git-changelog? What version of vue is it?

```json
  "dependencies": {
    "@iconify-json/octicon": "^1.1.55",
    "@nolebase/ui": "workspace:^",
    "colorette": "^2.0.20",
    "date-fns": "^3.6.0",
    "defu": "^6.1.4",
    "es-toolkit": "^1.13.1",
    "execa": "^8.0.1",
    "globby": "^14.0.1",
    "gray-matter": "^4.0.3",
    "less": "^4.2.0",
    "uncrypto": "^0.1.3",
    "vitepress": "^1.2.3"
  },
  "devDependencies": {
    "@rollup/plugin-yaml": "^4.1.2",
    "builtin-modules": "^3.3.0"
  }
```

root dep vue=3.4.30

This could be an issue with the vue version or the pnpm link, as it is normal in the workspace. We can put that aside for a moment, and I'll take some time to look into the question. Thanks!

nekomeowww commented:
> What is the resolved Vue version inside of the VitePress module?

3.4.31

> What is the resolved Vue version inside of the NolebaseGitChangelog module?

3.4.30

Then there is a mismatch issue. We can simply solve this issue by manually install `vue` as dependencies for all Vue related plugins...

Surely the `pnpm` team may somehow understand this issue better than me/us (maybe?). I'd suggest you to contact or file a issue to their repo. (If you have time to craft minimum repo, it's quite hard since you will need to projects side by side and keeping one as monorepo/workspace, and normal package for the other.

nekomeowww reviewed (approved):
Thanks for the hard work. LGTM!

nekomeowww commented:
Feel free to squash and merge at any time you feel it's ready. Much appriciated, again.

I have to focus something else since I got several other things that needed to be done and investigated on.

northword commented:
> Then there is a mismatch issue. 

Yes, it was a mismatch, once I updated both sides of Vue to 3.4.31 and VitePress to 1.3.1 (in fact, I updated all the dependencies to the latest version), the problem was solved, thanks for the solution!
